### DISCUSSION TITLE
Cue template to Nickel

### DISCUSSION THREAD

**Discussion by GTrunSec** [tweag/nickel#1717] [Ideas]
Is it possible to have a feature like https://cuetorials.com/first-steps/generate-all-the-things/

**Comment by vkleen** [comment]
I'm not sure I understand exactly what you're asking for. But Nickel has string interpolation facilities and there is a `raw` export format available. Additionally with the latest release we can export specific fields in a record. I.e. `nickel export -f raw --field output` on
```
{
  variable = 4,
  output = m%"
    The variable is %{std.string.from variable}
  "%
}
```
produces
```
❯ nickel export -f raw --field output test.ncl
The variable is 4
```

**Reply by GTrunSec** [reply]
You are correct, I may have overcomplicated the problem. just using string interpolation is enough same as concatString.

**Reply by yannham** [reply]
You can even set the variable from the command line:

```nickel
{
  variable | default = 4,
  output = m%"
    The variable is %{std.string.from variable}
  "%
}
```

```console
$ nickel export config.ncl --format raw --field output -- variable=10
The variable is 10
```

**Reply by GTrunSec** [reply]
@yannham 
Thanks for your tip,  I would like to know if I can use `cat json.file` to a variable.

```nickel
# works
nickel export ./nickel/test.ncl --format json --field data -- data='import "./t.json"'
{
  "a": "s"
}
# how about 
cat t.json | nickel export ./nickel/test.ncl --format json --field data -- data=<possible>
{
  "a": "s"
}
```

**Reply by yannham** [reply]
Sorry @GTrunSec, I just only see your comment. Currently there's no special way to do that, no. Maybe `xargs` could help but I think it has some limit on the size, so it might not be a great solution. Because the value of fields must be valid Nickel syntax, it could be possible to reserve some special marker - as long as it's not valid Nickel - to say that we want to take some field from stdinput, like `<stdin>` or `-`. If you think it's a valuable addition, I would recommend to open an issue for it, such that we don't lose track of it.
